indexing-service/src/main/java/org/apache/druid/indexing/common/KillTaskReport.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.indexing.common;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class KillTaskReport implements TaskReport
+{
+  public static final String REPORT_KEY = "killUnusedSegments";
+
+  private final String taskId;
+  private final Stats stats;
+
+  @JsonCreator
+  public KillTaskReport(
+      @JsonProperty("taskId") String taskId,
+      @JsonProperty("payload") Stats stats
+  )
+  {
+    this.taskId = taskId;
+    this.stats = stats;
+  }
+
+  @Override
+  @JsonProperty
+  public String getTaskId()
+  {
+    return taskId;
+  }
+
+  @Override
+  public String getReportKey()
+  {
+    return REPORT_KEY;
+  }
+
+  @Override
+  @JsonProperty
+  public Object getPayload()
+  {
+    return stats;
+  }
+
+  public static class Stats
+  {
+    private final int numSegmentsKilled;
+    private final int numBatchesProcessed;
+    private final int numSegmentsMarkedAsUnused;
+
+    @JsonCreator
+    public Stats(
+        @JsonProperty("numSegmentsKilled") int numSegmentsKilled,
+        @JsonProperty("numBatchesProcessed") int numBatchesProcessed,
+        @JsonProperty("numSegmentsMarkedAsUnused") int numSegmentsMarkedAsUnused
```

Comment:
Since this is being deprecated in the kill task, should we omit it in task report?

Reply:
I was thinking the same but then I decided to keep it for now so that if anyone enables the `markAsUnused` flag, they can find out from the report if it had any effect.

Reply:
if we add it though, can we easily remove later on as I intend we'll want to do, without breaking the api contract? Maybe we can mark with as nullable? So that we can remove later on? idk, what do you think?

Reply:
Yeah, fair point, old code would not be able to read future reports which don't have the `numSegmentsMarkedAsUnused` field. Let me mark this as `@Nullable`.

Reply:
Updated.